src/resolvers/namespace.resolver.ts

```diff
@@ -48,3 +51,23 @@ export class NamespaceIdResolver implements Resolver {
         return new NamespaceId([namespaceIdUInt64.lower, namespaceIdUInt64.higher]);
     }
 }
+
+/**
+ * Root namespace resolver
+ */
+export class RootNamespaceResolver implements Resolver {
```

Comment:
I think the `RootNamespaceResolver` name is a bit confusing because of the other resolvers behaviours
At first sight, I would expect `RootNamespaceResolver` to return a root namespace object

I propose to rename it to `NamespaceTypeResolver` and to have it return a `NamespaceRegistrationType`

Then in the command, you could do like:
`if (namespaceType === NamespaceRegistrationType.RootNamespace) `...

Reply:
Good catch, should be modified in the latest commit.